.github/workflows/create-condalock-file.yml

```diff
@@ -0,0 +1,88 @@
+name: Conda lock file creation
+
+on:
+  push:
+    branches:
+    - main
+    - condalock
+  schedule:
+    - cron: '0 4 */10 * *'
+
+jobs:
+  create-lock-file:
+    name: Create conda lock file for latest Python
+    runs-on: 'ubuntu-latest'
+    steps:
+      - uses: actions/checkout@v2
+      - uses: conda-incubator/setup-miniconda@v2
+        with:
+          auto-update-conda: true
+          activate-environment: esmvaltool-fromlock
+          python-version: "3.10"
+          miniconda-version: "latest"
+          channels: conda-forge
+      - name: Show conda config
+        shell: bash -l {0}
+        run: |
+          conda info
+          conda list
+          conda config --show-sources
+          conda config --show
+      - name: Python info
+        shell: bash -l {0}
+        run: |
+          which python
+          python --version
+      - name: Install conda-lock
+        shell: bash -l {0}
+        run: conda install -y conda-lock
+      - name: Create conda lock file for linux-64
+        shell: bash -l {0}
+        run: conda-lock lock --platform linux-64 -f environment.yml --mamba
+      - shell: bash -l {0}
+        run: conda --version
+      - shell: bash -l {0}
+        run: which python
+      - shell: bash -l {0}
+        run: python -V
+      - shell: bash -l {0}
+        run: conda create --name esmvaltool-fromlock --file conda-linux-64.lock
+      - shell: bash -l {0}
+        run: conda install pip
+      - shell: bash -l {0}
+        run: which python
+      - shell: bash -l {0}
+        run: pip --version
+      - shell: bash -l {0}
+        run: pip install -e .[develop]
+      - shell: bash -l {0}
+        run: esmvaltool --help
+      - shell: bash -l {0}
+        run: esmvaltool version
+      - shell: bash -l {0}
+        run: pytest -n 2 -m "not installation"
```

Comment:
Do we need to put every single line into its own shell?

Also, is it not necessary to activate the newly created environment? When does conda switch to `esmvaltool-fromlock`?

Reply:
about the shell encasisng - that's my personal preference (well, how I got used to this stuff), so now I OCD quickly if I don't see each line with its own shell call :grin: Very good question about the `esmvaltool-fromlock` environment! Currently `conda-incubator/setup-miniconda@v2` doesn't allow the creation of a conda env from a lock file, so I am going around this issue by asking for an env to be created at initial point, then I am re-creating that env explicitly with `conda create --name esmvaltool-fromlock --file conda-linux-64.lock` which is in fact the equivalent of an update of that already created env (note that conda env update or conda activate are NOT supported commands by the incubator)

Reply:
I don't know about incubator, but I do know that `conda activate evt` runs the activation scripts under `$CONDA_PREFIX/etc/conda/activate.d`, which for us at least contains scripts for gdal, udunits2, proj4, and ncl. I would not leave out `conda activate` light-heartedly.

Reply:
well yes, that's the whole thing - the incubator is not allowing messing with the conda path and such - but not a worry, all's done exactly as if `activate` was called, there are no differences here

Reply:
the activation is done implicitly through the incubator, I meant, if you look at the Run conda-incubator/setup miniconda2 step in the Action https://github.com/ESMValGroup/ESMValTool/runs/4843383653?check_suite_focus=true - you see Updating 'esmvaltool-fromlock' env from conda create (simple)...

Reply:
Iff activation happens, that surely is at the incubator step, no? In other words, before the activation scripts are installed.
Also, the documentation of incubator actually shows how to use lockfiles: https://github.com/conda-incubator/setup-miniconda#example-7-lockfiles

Reply:
yes that doesn't work for us, good find tho - I was well happy when I found too, only to be sad it don't work; we should tell those guys about it not working at some point

Reply:
So it seems to me that the question of activation is still open.  Where and how are the activation scripts that are installed in line 49 called?

Reply:
the environment is already created at the beginning, at the `activate-environment: esmvaltool-fromlock` bit in the incubator, we are merely adding dependencies to it by doing the `conda create --name esmvaltool-fromlock --file conda-linux-64.lock` call - I guess/hope that all relevant executables are then placed in the right place, I think if anything goes awry then it'd be caught in the verification/running tests step

Reply:
Yes, exactly, we are adding dependencies, which bring new activation scripts that will certainly be executed on the next call to `conda activate`. Are they executed in another way? How? Where?

Reply:
hmmm - indeed - good Columbo attitude, man - `conda activate` is forbidden inside the workflow with the incubator so I *believe* upon "creation" of the env from the condalock file, which is in effect, an `update` of an existing active env, new things get automatically picked up - just like after a `conda update -n ENV -f EILE` one doesn't have to activate after...but TBF am not 100% sure, might be worth asking the conda incubator guys, all I can see is that stuff (incl Julia/R stuff) works in the GA test, and the user instructions do it properly anyway, with actual activation after env creation

Reply:
Well, I am asking precisely because I have had problems with a lack of activation script execution after `conda update` in an active environment. Not sure how to test this though.

Reply:
sorry, browser hiccup - didnt mean to resolve the conversation!

Reply:
yeah fair do's - but I reckon we're overly complicating the issue specific to the GA test only - it works fine, don't fix it if ain't broken :grin: The important stuff ie us installing from a conda lock file is done proper. Plus, the conda lock builder don't need no activation, it's working separately

Reply:
I think we are opening up ourselves to problems down the line. But not big problems. And I definitely have earned a "told-you-so" for that case, so go ahead.

Reply:
ahaha, good one, dude! Well, yes, I totally agree with you - but heyho, we'll have to switch all this to a dedicated env creation+activation fromlock straight from the incubator anyway, whenever they allow full functionality, plus to mamba anyway, so this is probably just an interim solution (half baked, but it worka :grin: )